source/common/logger/ur_logger_details.hpp

```diff
@@ -70,9 +114,14 @@ class Logger {
         }
     }
 
+    void setLegacySink(std::unique_ptr<logger::Sink> legacySink) {
+        this->legacySink = std::move(legacySink);
+    }
+
   private:
     logger::Level level;
     std::unique_ptr<logger::Sink> sink;
+    std::unique_ptr<logger::Sink> legacySink;
```

Comment:
why do we need a legacySink that's separate from the sink? Are they going to be used at the same time?

Reply:
Technically we don't need both - but we have to know if an legacy one is set to skip ifs checking log levels. IF you preffer i can replace legacy Sink with a Boolean value.

Reply:
yea, a bool would be better.

Reply:
Done